odo/backends/tests/test_csv.py

```diff
@@ -314,3 +314,12 @@ def test_infer_header():
 
 def test_csv_supports_sep():
     assert CSV('foo.csv', sep=';').dialect['delimiter'] == ';'
+
+
+def test_csv_to_compressed_csv():
+    with tmpfile('.csv') as fn:
+        with open(fn, 'w') as f:
+            f.write('a,1\nb,2\nc,3')
+        with tmpfile('.csv.gz') as gfn:
+            result = odo(fn, gfn)
+            assert odo(result, list) == odo(fn, list)
```

Comment:
Do these need to be `seek`ed back to the start? I don't know how odo handles opened files. Just worried you may be comparing 2 empty lists.